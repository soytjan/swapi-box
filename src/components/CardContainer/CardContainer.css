@import url("https://fonts.googleapis.com/css?family=Bungee+Hairline|Bungee+Outline|Roboto|Roboto+Condensed");
.card-container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 20px;
  background-color: #2B2B2D; }
  .card-container h2 {
    grid-column: 1/7;
    text-transform: capitalize;
    font-family: "Bungee Hairline", cursive;
    font-size: 1.5em;
    color: white; }

.loading {
  grid-column: 1/7; }
