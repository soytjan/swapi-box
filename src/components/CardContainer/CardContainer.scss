@import "../../variables.scss";

.card-container {
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 20px;
  background-color: $box-bg-color;

  h2 {
    grid-column: 1/7;
    text-transform: capitalize;
    font-family: $header-box-font;
    font-size: 1.5em;
    color: white;
  }
}