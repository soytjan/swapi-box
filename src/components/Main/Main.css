@import url("https://fonts.googleapis.com/css?family=Bungee+Inline|Roboto|Roboto+Condensed");
main {
  background-color: #434346;
  padding: 10px 20px;
  margin: 0 30px; }

.card-box {
  min-height: 700px;
  background-color: #434346; }
