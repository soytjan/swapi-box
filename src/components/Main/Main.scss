@import "../../variables.scss";

main {
  background-color: $main-bg-color;
  padding: 10px 20px;
}

.card-box {
  min-height: 700px;
  background-color: $box-bg-color;
}