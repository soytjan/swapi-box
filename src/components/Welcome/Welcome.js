import React from 'react';
import PropTypes from 'prop-types';
import ScrollContainer from '../ScrollContainer/ScrollContainer';
import './Welcome.css';

const Welcome = ({film}) => {
  return (
    <div className='welcome'>
      I'm Welcome
      <ScrollContainer {...film} />
    </div>
  )
}

Welcome.propTypes = {
  onClick: PropTypes.func,
};

Welcome.propTypes = {
  film: PropTypes.object,
};

export default Welcome;