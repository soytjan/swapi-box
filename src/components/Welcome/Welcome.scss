.welcome {
  width: 100%;
  height: 100%;
  overflow: hidden;
  background: #000;
  padding: 0 20%;
}