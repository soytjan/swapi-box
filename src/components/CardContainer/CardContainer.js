import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import './CardContainer.css'

const CardContainer = ({data, type, onFavClick, favorites}) => {
  let renderedCards;
  
  if (type === 'favorites' && data.length === 0) {
    renderedCards = (<p className='loading'>You have no favorites selected!</p>)
  } else if (data.length === 0) {
    renderedCards = (<div className='loading'>Loading...</div>)
  } else {
    renderedCards = data.map((info, index) => {
      const isFavorited = favorites.includes(info) ? 'favorited' : '';
    
      return <Card info={info} type={type} isFavorited={isFavorited} onFavClick={onFavClick} key={`${index}${Date.now()}`} />
    })
  }

  return(
    <section className="card-container">
      <h2>{type}</h2>
      {renderedCards}
    
    </section>
  )
}

CardContainer.propTypes = {
  data: PropTypes.array,
  type: PropTypes.string,
  onFavClick: PropTypes.func,
  favorites: PropTypes.array,
};

export default CardContainer;