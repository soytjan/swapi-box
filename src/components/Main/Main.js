import React from 'react';
import { Switch, Route } from 'react-router-dom';
import People from '../People/People';
import Welcome from '../Welcome/Welcome';
import CardContainer from '../CardContainer/CardContainer';
import './Main.css';

const Main = ({people, vehicles, planets}) => {
  return (
    <main>
      <div className='card-box'>
      <Switch>
        <Route exact path='/' render={() => (<Welcome />)} />
        <Route path='/people' render={() => (<CardContainer data={people}/>)} />
        <Route path='/vehicles' render={() => (<CardContainer data={vehicles} />)} />
        <Route path='/planets' render={() => (<CardContainer data={planets} />)} />
        <Route path='/favorites' render={() => (<CardContainer />)} />
      </Switch>
      </div>
    </main>
  )
}

export default Main;