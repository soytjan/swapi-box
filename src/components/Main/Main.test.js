import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Welcome from '../Welcome/Welcome';
import CardContainer from '../CardContainer/CardContainer';
import Main from './Main';

describe('Main', () => {
  it.skip('should match snapshot', () => {
    const mockFilm = {title: "Revenge of the Sith", crawl: "War! The Republic is crumbling under attacks by tâ€¦ate mission to rescue the captive Chancellor....", releaseDate: "2005-05-19"};
    const mockPeople = [{ name: "Luke Skywalker",homeworld: "Tatooine", population: 200000,species: "Human" }];
    const mockPlanet = [{name: "Coruscant", terrain: "cityscape, mountains", population: "1000000000000", climate: "temperate", residents: ["Finis Valorum", "Adi Gallia", "Jocasta Nu"]}];
    const mockVehicle = [{ "name": "Sand Crawler","model": "Digger Crawler", "passengers": "30", "vehicleClass": "wheeled"}];
    const mockFavClick = jest.fn();
    const renderedComponent = shallow(<Main film={mockFilm} people={mockPeople} vehicles={mockVehicle} planets={mockPlanet} onFavClick={mockFavClick} favorites={[]} />)

    expect(renderedComponent).toMatchSnapshot();
  });

  it('should load Welcome page by default', () => {
    const renderedComponent = mount(
      <MemoryRouter intialEntries={[ '/' ]}>
        <Main />
      </MemoryRouter>
    );

    expect(renderedComponent.find(Welcome)).toHaveLength(1);
    expect(renderedComponent.find(CardContainer)).toHaveLength(0);
  });
})