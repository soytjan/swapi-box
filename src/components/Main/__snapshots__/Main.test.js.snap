// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    favorites={Array []}
    film={
        Object {
            "crawl": "War! The Republic is crumbling under attacks by tâ€¦ate mission to rescue the captive Chancellor....",
            "releaseDate": "2005-05-19",
            "title": "Revenge of the Sith",
          }
    }
    onFavClick={[Function]}
    people={
        Array [
            Object {
              "homeworld": "Tatooine",
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
          ]
    }
    planets={
        Array [
            Object {
              "climate": "temperate",
              "name": "Coruscant",
              "population": "1000000000000",
              "residents": Array [
                "Finis Valorum",
                "Adi Gallia",
                "Jocasta Nu",
              ],
              "terrain": "cityscape, mountains",
            },
          ]
    }
    vehicles={
        Array [
            Object {
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicleClass": "wheeled",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-box"
>
        <Switch>
                <Route
                        exact={true}
                        path="/"
                        render={[Function]}
                />
                <Route
                        path="/people"
                        render={[Function]}
                />
                <Route
                        path="/vehicles"
                        render={[Function]}
                />
                <Route
                        path="/planets"
                        render={[Function]}
                />
                <Route
                        path="/favorites"
                        render={[Function]}
                />
        </Switch>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    exact={true}
                    path="/"
                    render={[Function]}
          />
          <Route
                    path="/people"
                    render={[Function]}
          />
          <Route
                    path="/vehicles"
                    render={[Function]}
          />
          <Route
                    path="/planets"
                    render={[Function]}
          />
          <Route
                    path="/favorites"
                    render={[Function]}
          />
</Switch>,
        "className": "card-box",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              exact={true}
              path="/"
              render={[Function]}
/>,
            <Route
              path="/people"
              render={[Function]}
/>,
            <Route
              path="/vehicles"
              render={[Function]}
/>,
            <Route
              path="/planets"
              render={[Function]}
/>,
            <Route
              path="/favorites"
              render={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/people",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/vehicles",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/planets",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-box"
>
          <Switch>
                    <Route
                              exact={true}
                              path="/"
                              render={[Function]}
                    />
                    <Route
                              path="/people"
                              render={[Function]}
                    />
                    <Route
                              path="/vehicles"
                              render={[Function]}
                    />
                    <Route
                              path="/planets"
                              render={[Function]}
                    />
                    <Route
                              path="/favorites"
                              render={[Function]}
                    />
          </Switch>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        exact={true}
                        path="/"
                        render={[Function]}
            />
            <Route
                        path="/people"
                        render={[Function]}
            />
            <Route
                        path="/vehicles"
                        render={[Function]}
            />
            <Route
                        path="/planets"
                        render={[Function]}
            />
            <Route
                        path="/favorites"
                        render={[Function]}
            />
</Switch>,
          "className": "card-box",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                exact={true}
                path="/"
                render={[Function]}
/>,
              <Route
                path="/people"
                render={[Function]}
/>,
              <Route
                path="/vehicles"
                render={[Function]}
/>,
              <Route
                path="/planets"
                render={[Function]}
/>,
              <Route
                path="/favorites"
                render={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/people",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/vehicles",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/planets",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
